info/guide/guide-chapter-3.md

```diff
@@ -345,19 +356,80 @@ try {
 
 Line length should be less than 120 symbols. Otherwise, it should be split.
 
-If `complex property` initializing is too long, it should be split:
+if the initializer of a `complex property` is too long, it should be split into multiple lines using the operators grouped in the following precedence order:
+1. Logic Binary Expression (`&&` , `||`)
+2. Comparison Binary Expression (`>` , `<` , `==` , `>=` , `<=` , `!=`, `===`, `!==`)
+3. Other types (Arithmetical and Bitwise operation) (`+` , `-` , `*` , `/` , `%` , `>>` , `<<` , `&` , `|` , `~` , `^` , `>>>` , `<<<` , `*=` , `+=` , `-=` , `/=` , `%=` , `++` , `--` , `in` , `!in` etc)
 
 **Invalid example:**
 ```kotlin
 val complexProperty = 1 + 2 + 3 + 4
 ```
 **Valid example:**
 ```kotlin
-val complexProperty = 1 + 2
-+ 3 + 4
+val complexProperty = 1 + 2 +
+    3 + 4
+```
+
+**Invalid example:**
+```kotlin
+val complexProperty = (1 + 2 + 3 > 0) && ( 23 * 4 > 10 * 6)
+```
+**Valid example:**
+```kotlin
+val complexProperty = (1 + 2 + 3 > 0) &&
+    (23 * 4 > 10 * 6)
+```
+
+A long expression which should be split into two lines before the `Elvis Operator` (`?:`):
+
+**Invalid example:**
+```kotlin
+val value = first ?: second
+```
+
+**Valid example:**
+```kotlin
+val value = first
+    ?: second
+```
+
+A long line in a `Dot Qualified Expression` or a `Safe Access Expression`:
+
+**Invalid example:**
+```kotlin
+val value = This.Is.A.Very.Long.Dot.Qualified.Expression
+```
+**Valid example:**
+```kotlin
+val value = This.Is.A.Very.Long
+    .Dot.Qualified.Expression
+```
+
+**Invalid example:**
+```kotlin
+val value = This.Is?.A?.Very?.Long?.Safe?.Access?.Expression
+```
+**Valid example:**
+```kotlin
+val value = This.Is?.A?.Very?.Long
+    ?.Safe?.Access?.Expression
 ```
 
-If `annotation` is too long, it also should be split:
+A long `function call arguments`:
```

Comment:
"A" implies a singular form. "Call arguments" is plural.

> "A long list of `function call arguments`".

Reply:
Done [a77aa42](https://github.com/saveourtool/diktat/pull/1384/commits/a77aa42cf1556d81c72b89e2bd01872de062d7d2)